src/doc_builder/commands/push.py

```diff
@@ -0,0 +1,208 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+from pathlib import Path
+from time import time
+from typing import List, Optional, TypedDict
+
+import requests
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+
+def get_head_oid(repo_id: str, token: str, branch: Optional[str] = "main") -> str:
+    """
+    Returns last commit sha from repostory `repo_id` & branch `branch`
+    """
+    res = requests.get(
+        f"https://api.github.com/repos/{repo_id}/git/refs/heads/{branch}",
+        headers={"Authorization": f"bearer {token}"},
+    )
+    if res.status_code != 200:
+        raise Exception(f"get_head_oid failed: {res.message}")
+
+    res_json = res.json()
+    head_oid = res_json["object"]["sha"]
+    return head_oid
+
+
+class FileAddition(TypedDict):
+    path: str
+    contents: int
```

Comment:
`class FileAddition(TypedDict)` is purely for typing reasons using [TypedDict](https://peps.python.org/pep-0589/)
if I use dataclass, then, I have to change line below to use the dataclass as in `additions.append(new FileAddition(...))`, which would break the graphql https://github.com/huggingface/doc-builder/blob/59e8f863bc878ed2b62403500c6bcd5f26003710/src/doc_builder/commands/push.py#L62

Reply:
Oh I think you were actually using it.
This is Python and we don't use type checker on any of our repos. I don't think introducing this helps in any way, so I'd remove it as well as the type annotations below (`Dict` is enough honestly)

Also this is only available in Python 3.8+ unless some extra package is installed, which the setup of `doc-builder` doesn't have.

Reply:
handled in https://github.com/huggingface/doc-builder/pull/214/commits/5257820158750fe41371a025d5d16501c90ea2f7